### PR TITLE
Content Build JSON Comparison Script

### PR DESCRIPTION
**Pull request by theodur**
## Description
This script compares the data of the CMS export and GraphQL nodes, while providing multiple options for doing so.

## Testing done
Tested locally on various entities.

## Screenshots
**Script help menu**
Screen Shot 2021-01-20 at 5 51 30 PM

**Running script on specific entity**
Screen Shot 2021-01-20 at 5 52 50 PM

**Running script on multiple node types**
Screen Shot 2021-01-20 at 5 54 49 PM

**Running script on all node entities**
Screen Shot 2021-01-20 at 5 58 01 PM

## Acceptance criteria
- [ ] The comparison script should accurately compare the data of GraphQL and transformed CMS Export objects

## Definition of done
- [ ] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/cms/compare-json.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+      const ent = rawEntities.find(
+        entity =>
+          entity.baseType === 'node' &&
+          parseInt(entity.nid[0].value, 10) ===
+            parseInt(node.parent_id[0].value, 10),
+      );
+      getParentNode(ent, rawEntities, parent);
+    }
+  } else {
+    // TODO: Add support for deeper nested paragraphs, and other entity types
```
Comment (timwright12) [inline script/cms/compare-json.js:142]: @theodur  Do we need this for initial release? If not, should we make a ticket for it?

File: script/cms/compare-json.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        fs.writeFileSync(
+          path.join(__dirname, `../../content-object-diff.json`),
+          JSON.stringify(diff),
+        );
+        console.log(`${diff.length} differences: './content-object-diff.json'`);
+      } else {
+        console.log(`No differences found!`);
+      }
+    } else {
+      console.log('Node not present in .cache/localhost/drupal/pages.json');
```
Comment (timwright12) [inline script/cms/compare-json.js:257]: Will this always be /localhost/ ?

File: script/cms/compare-json.js (hunk 1 above)
Comment (theodur) [reply script/cms/compare-json.js:142]: I don't think we need this for the initial release. I can make a ticket for it and add it to the backlog.

File: script/cms/compare-json.js (hunk 2 above)
Comment (theodur) [reply script/cms/compare-json.js:257]: Nice catch, this won't always be localhost. Forgot to update this after working on this locally.

File: script/cms/compare-json.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+// Array of nodes from pages.json
+const graphQL = JSON.parse(
+  fs.readFileSync(
+    path.join(__dirname, `../../.cache/${buildtype}/drupal/pages.json`),
+  ),
+);
+
+const assembleEntityTree = assembleEntityTreeFactory(exportDir);
+
+if (help) {
```
Comment (billfienberg) [inline script/cms/compare-json.js:61]: Should we move the `help` check before the `JSON.parse` and tree assembly? If I'm reading it correctly, it seems like 1. adding the `help` arg would still parse JSON (which seems unexpected), and 2. the `help` command could fail if there was an error during the parse or assembly. 

Does my understanding makes sense, or am I missing something?

File: script/cms/compare-json.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+    // eslint-disable-next-line no-param-reassign
+    parent.node = node;
+  }
+
+  if (parent.path === '') {
+    if (node.parent_type[0].value === 'node') {
+      // eslint-disable-next-line no-param-reassign
+      parent.path += node.parent_field_name[0].value;
+      const ent = rawEntities.find(
+        entity =>
```
Comment (billfienberg) [inline script/cms/compare-json.js:142]: Do you think this function body could be more readable? 

For example:
```
entity=> {
  if (entity.baseType === 'node') {
    return parseInt(entity.nid[0].value, 10) === parseInt(node.parent_id[0].value, 10)
  }
  return false
}
```

File: script/cms/compare-json.js (hunk 3 above)
Comment (theodur) [reply script/cms/compare-json.js:47]: That's a good point. I just pushed a commit that moved the `help` check up.

**Comment by billfienberg** [comment]
Relevant links:
- https://dsva.slack.com/archives/G016SUY8AH5/p1611249196005100
- https://dsva.slack.com/archives/G016SUY8AH5/p1611337819008200

File: script/cms/compare-json.js (hunk 4 above)
Comment (theodur) [reply script/cms/compare-json.js]: Sure, I just pushed a commit with the second variation 🙂.

File: script/cms/compare-json.js (hunk 2 above)
Comment (theodur) [reply script/cms/compare-json.js]: I updated this in a previous commit

**Comment by timwright12** [comment]
LGTM

**Review by billfienberg — APPROVED** [review]

**Pull request merged**